### ISSUE TITLE
Webextension compatible

### ISSUE THREAD

**Issue by Dimas-sc** [schomery/tab-memory-usage#20]
Will this addon be compatible with Webextensions / Firefox 57?

**Comment by rshadow** [comment]
I'm using firefox 57. This plugin is broken :'(

**Comment by bunnybooboo** [comment]
@schomery there's been no activity on this add-on for a year. Do you currently have a roadmap to implement WebExtensions?

**Comment by schomery** [comment]
There is no API to read Tab's memory as far as I know! So there is no way to port this to WebExtension.

**Comment by Dimas-sc** [comment]
@schomery is there any request on bugzilla we can vote?

**Comment by schomery** [comment]
I don't think so. Though I havent recently searched the Bugzilla. If you found one please share here.

**Comment by yfdyh000** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1296898

**Comment by schomery** [comment]
Thanks @yfdyh000

**Comment by bunnybooboo** [comment]
The discussions across those bugs look somewhat encouraging.

**Comment by schomery** [comment]
We need something similar to https://developer.chrome.com/extensions/processes

**Comment by bunnybooboo** [comment]
Hmm. @schomery have you also seen https://bugzilla.mozilla.org/show_bug.cgi?id=1351346
I highly suggest jumping in on either of those bugs to step up as the experimental webextension they're talking of. If you might have time to further develop?

**Comment by rshadow** [comment]
https://developer.chrome.com/extensions/processes#method-getProcessInfo with `includeMemory`?

**Comment by schomery** [comment]
@rshadow yes. 

@bunnybooboo I am not really into it. I've reported several bugs in the past and usually, they go nowhere! When the API is ready in Chrome (not as an experimental one), I'll start porting the extension.

**Comment by bunnybooboo** [comment]
No worries @schomery My concern is that without a PoC the API might never happen.

**Comment by lxvi** [comment]
In addition to memory, cpu usage would be phenom! (yeah, this is a thinly disguised "bump" ;^) )

**Comment by garywill** [comment]
Hello guys,
I saw you were discussing on this. 
Now Firefox has `about:performance`
And, I did some modification on Mozilla's code to show CPU and memory bar on each tab, as userChrome script (https://github.com/garywill/firefoxtaskmonitor)
It's in beta but it works.
See screenshot here: https://www.reddit.com/r/FirefoxCSS/comments/ehkf4r/taskmonitor_show_cpu_memory_bar_per_tab_and_all/
